app.py

```diff
@@ -1,67 +1,129 @@
-    # long is not negative here because "TOP" in the request accounts for this.
-
-    # Fetching weather data
-    # Hardcoding to the correct weather station
-    weather_data = requests.get(
-        f'https://api.weather.gov/gridpoints/OKX/33,35/forecast'
+def getContext(lat, lng, default_forecast, timestamp):
+    area_data = requests.get(
+        f'https://api.weather.gov/points/{lat},{lng}'
     ).json()
+    
+    cwa, gridX, gridY = area_data['properties']['cwa'], area_data['properties']['gridX'], area_data['properties']['gridY']
 
-    # Period wise data
-    periodic_data = weather_data['properties']['periods']
-
-    latitude, longitude = weather_data['geometry']['geometries'][0]['coordinates'][
-        0], weather_data['geometry']['geometries'][0]['coordinates'][1]
-
-    # Fetching area data
-    area_data = requests.get(
-        f'https://api.weather.gov/points/{longitude},{latitude}'
+    weather_data = requests.get(
+        f'https://api.weather.gov/gridpoints/{cwa}/{gridX},{gridY}/forecast'
     ).json()
 
-    print(area_data)
+    try:
+        periodic_data = weather_data['properties']['periods']
+    except:
+        context = {
+            'error': True,
+            'title': weather_data['title'],
+            'description': weather_data['detail']
+        }
+        return render_template('index.html', **context)
 
-    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state']
+    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state'] +['',' (Default)'][default_forecast]
 
     context = {
         'timestamp': timestamp,
         'periodic_data': periodic_data,
         'areaDescription': areaDescription
     }
 
-    return render_template('index.html', **context)
+    return context
+
+@app.route('/')
+@app.route('/index')
+@app.route('/<zip_code>')
+def index(zip_code=None):
+    
+    timestamp = datetime.datetime.now()
+
+    if zip_code != None and re.search(r'.*(\d{5}?)$', zip_code):
+        if cache.get(CACHE_1_ZIP) is not None and cache.get(CACHE_1_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_1_ZIP)['context'])
+        elif cache.get(CACHE_2_ZIP) is not None and cache.get(CACHE_2_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_2_ZIP)['context'])
+        elif cache.get(CACHE_3_ZIP)  is not None and cache.get(CACHE_3_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_3_ZIP)['context'])
+        else:
+            lat, lng, default_value = 0, 0, False
+            
+            # zipcode-lat-lng.csv
+            # https://gist.github.com/abatko/ee7b24db82a6f50cfce02afafa1dfd1e
+            with open('./assets/zipcode-lat-lng.csv', 'r') as f:
```

Comment:
The zip code values aren't returning correct weather location data. For comparison, I'm trying to get `10025` to match.

https://forecast.weather.gov/MapClick.php?lat=40.7999209&lon=-73.96831020000002&site=all&smap=1&searchresult=New%20York%2C%20NY%2010025%2C%20USA

However, when I use zipcodes from the file, I get `West New York, NJ`

10018 should also be New York, but it's returning NJ. 

The forecasts are *very* close but there are differences.
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/2412992/67517404-8afb3600-f670-11e9-8e4b-2024a764d9fe.png">

This is likely an original bug in my default lat/long having it return a *close* New Jersey weather station instead of the specific New York one.